evalml/update_checker.py

```diff
@@ -0,0 +1,9 @@
+from pkg_resources import iter_entry_points
```

Comment:
@jeff-hernandez to date we've reserved the top directory level of the `evalml` folder for modules only. Could you please move this file to `evalml/utils/`?

Reply:
Yep, moved